webapp/packages/plugin-data-viewer/src/ValuePanelPresentation/ImageValue/ImageValuePresentation.tsx

```diff
@@ -82,21 +83,19 @@ export const ImageValuePresentation: TabContainerPanelComponent<IDataValuePanelP
 
           return this.formatAction.get(this.selectedCell);
         },
-        get src(): string | null {
+        get src(): string | Blob | null {
           if (isResultSetBlobValue(this.cellValue)) {
             // uploaded file preview
-            return URL.createObjectURL(this.cellValue.blob);
+            return this.cellValue.blob;
           }
 
           if (this.staticSrc) {
```

Comment:
probably we want this condition to be at the end of the function
so if we have some blobs cached we return it first
and only then if there are nothing to give from cache we can give a link to download

Reply:
priority is:
1. changed value (preview)
2. available src data (we don't need to have cache in this case)
3. available cache if src is not available